sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -99,32 +104,41 @@ private async ValueTask<AccessToken> GetTokenImplAsync(bool async, TokenRequestC
 
             try
             {
-                AccessToken token;
-
-                if (_credential != null)
-                {
-                    token = async ? await _credential.GetTokenAsync(requestContext, cancellationToken).ConfigureAwait(false) : _credential.GetToken(requestContext, cancellationToken);
-                }
-                else
-                {
-                    token = await GetTokenFromSourcesAsync(async, requestContext, cancellationToken).ConfigureAwait(false);
-                }
+                AccessToken token = _credential != null
+                    ? await GetTokenFromCredentialAsync(async, requestContext, cancellationToken).ConfigureAwait(false)
+                    : await GetTokenFromSourcesAsync(async, requestContext, cancellationToken).ConfigureAwait(false);
 
                 return scope.Succeeded(token);
             }
-            catch (Exception e) when (!(e is CredentialUnavailableException))
+            catch (Exception e)
             {
-               throw scope.FailWrapAndThrow(new AuthenticationFailedException(UnhandledExceptionMessage, e));
+               throw scope.FailWrapAndThrow(e);
             }
         }
 
-        private async ValueTask<AccessToken> GetTokenFromSourcesAsync(bool async, TokenRequestContext requestContext, CancellationToken cancellationToken)
+        private async ValueTask<AccessToken> GetTokenFromCredentialAsync(bool async, TokenRequestContext requestContext, CancellationToken cancellationToken)
         {
-            int i;
+            try
+            {
+                return async
+                    ? await _credential.GetTokenAsync(requestContext, cancellationToken).ConfigureAwait(false)
+                    : _credential.GetToken(requestContext, cancellationToken);
+            }
+            catch (CredentialUnavailableException)
+            {
+                throw;
+            }
+            catch (Exception e)
```

Comment:
Why not leave this as a filter as it was before, to avoid catching at all.

~~~c#
catch (Exception e) when (!(e is CredentialUnavailableException))
~~~

Reply:
Fixed